docs/rfcs/rfc-034-generic-operators.md

```diff
@@ -0,0 +1,506 @@
+
+to the types needed.
+
+This mechanism is adopted widely, for instance, consider the aggregation filters:
+
+```graphql
+input PostLikesNodeAggregationWhereInput {
+    AND: [PostLikesNodeAggregationWhereInput!]
+    OR: [PostLikesNodeAggregationWhereInput!]
+    NOT: PostLikesNodeAggregationWhereInput
+    name_EQUAL: String
+    name_AVERAGE_EQUAL: Float
+    name_LONGEST_EQUAL: Int
+    name_SHORTEST_EQUAL: Int
+    name_GT: Int
+    name_AVERAGE_GT: Float
+    name_LONGEST_GT: Int
+    name_SHORTEST_GT: Int
+    name_GTE: Int
+    name_AVERAGE_GTE: Float
+    name_LONGEST_GTE: Int
+    name_SHORTEST_GTE: Int
+    name_LT: Int
+    name_AVERAGE_LT: Float
+    name_LONGEST_LT: Int
+    name_SHORTEST_LT: Int
+    name_LTE: Int
+    name_AVERAGE_LTE: Float
+    name_LONGEST_LTE: Int
+    name_SHORTEST_LTE: Int
+    someFloat_EQUAL: Float
+    someFloat_AVERAGE_EQUAL: Float
+    someFloat_MIN_EQUAL: Float
+    someFloat_MAX_EQUAL: Float
+    someFloat_SUM_EQUAL: Float
+    someFloat_GT: Float
+    someFloat_AVERAGE_GT: Float
+    someFloat_MIN_GT: Float
+    someFloat_MAX_GT: Float
+    someFloat_SUM_GT: Float
+    someFloat_GTE: Float
+    someFloat_AVERAGE_GTE: Float
+    someFloat_MIN_GTE: Float
+    someFloat_MAX_GTE: Float
+    someFloat_SUM_GTE: Float
+    someFloat_LT: Float
+    someFloat_AVERAGE_LT: Float
+    someFloat_MIN_LT: Float
+    someFloat_MAX_LT: Float
+    someFloat_SUM_LT: Float
+    someFloat_LTE: Float
+    someFloat_AVERAGE_LTE: Float
+    someFloat_MIN_LTE: Float
+    someFloat_MAX_LTE: Float
+    someFloat_SUM_LTE: Float
+}
+```
+
+And then these can be used as follow:
+
+**Posts where content contains Matrix**
+
+```graphql
+query Posts {
+    posts(where: { content_CONTAINS: "The Matrix" }) {
+        content
+    }
+}
+```
+
+**Posts with more than 10 likes**
+
+```graphql
+query Posts {
+    posts(where: { likesAggregate: { count_GT: 10 } }) {
+        content
+    }
+}
+```
+
+## Proposed Solution
+
+As part of the new API design, this RFC wants to discuss a new approach to access these operators.
+
+Is possible to reuse generic filters, for instance:
+
+```graphql
+input UserWhere {
+    OR: [UserWhere!]
+    AND: [UserWhere!]
+    NOT: PostWhere
+    someArr: [Int!]
+    someFloat: FloatWhere
+    name: StringWhere
+}
+
+input PostWhere {
+    OR: [PostWhere!]
+    AND: [PostWhere!]
+    NOT: PostWhere
+    content: StringWhere
+    likes: MoviesLikesWhere
+}
+
+input MoviesLikesWhere {
+    OR: [MoviesLikesWhere!]
+    AND: [MoviesLikesWhere!]
+    NOT: MoviesLikesWhere
+    aggregation: PostLikesAggregateInput
+    all: PostLikesConnectionWhere
+    none: PostLikesConnectionWhere
+    single: PostLikesConnectionWhere
+    some: PostLikesConnectionWhere
+    # For simplicity the standard filters are substituted by the connections filters as discussed in the API uniformity piece.
+}
+
+input IntListWhere {
+    OR: [IntListWhere!]
+    AND: [IntListWhere!]
+    NOT: IntListWhere
+    all: [Int!]
+    some: Int
```

Comment:
Let's change these back - I definitely don't think these make sense.

`all` is just `equals` and `some` just doesn't make sense at all here. I would have the following operations:

- `equals`
- `includes`
- `intersects` (future, let's leave this out for now)